File: genzura-web/remove_bg.py
import os
from PIL import Image

def process():
    try:
        from rembg import remove
    except ImportError:
        print("rembg is not installed. Please wait for pip install to finish.")
        return

    files_to_process = [
        "public/Genzura Logo.png",
        "public/Genzura full logo.png",
        "public/Genzura website header.png"
    ]

    for file_path in files_to_process:
        if not os.path.exists(file_path):
            print(f"File not found: {file_path}")
            continue
            
        print(f"Processing {file_path}...")
        try:
            input_image = Image.open(file_path)
            output_image = remove(input_image)
            output_image.save(file_path)
            print(f"Successfully processed and saved {file_path}")
        except Exception as e:
            print(f"Error processing {file_path}: {e}")

if __name__ == "__main__":
    process()

File: genzura-web/remove_bg.js
import { Jimp } from 'jimp';

async function removeBg(imagePath) {
    try {
        console.log(`Processing ${imagePath}...`);
        const image = await Jimp.read(imagePath);
        
        // Define tolerance (how close to white before making transparent)
        // Values from 0 to 255. High tolerance means strictly white.
        const tolerance = 240; 
        
        image.scan((x, y, idx) => {
            const r = image.bitmap.data[idx + 0];
            const g = image.bitmap.data[idx + 1];
            const b = image.bitmap.data[idx + 2];
            
            if (r >= tolerance && g >= tolerance && b >= tolerance) {
                // Set alpha to 0 (transparent)
                image.bitmap.data[idx + 3] = 0;
            }
        });
        
        await image.write(imagePath);
        console.log(`Successfully processed ${imagePath}`);
    } catch (err) {
        console.error(`Error processing ${imagePath}:`, err.message);
    }
}

async function main() {
    const files = [
        "public/Genzura Logo.png",
        "public/Genzura full logo.png",
        "public/Genzura website header.png"
    ];
    
    for (const f of files) {
        await removeBg(f);
    }
}

main();
